services/publicapi/test/get_transaction_test.go

```diff
@@ -0,0 +1,93 @@
+package test
+
+import (
+	"context"
+	"github.com/orbs-network/orbs-network-go/crypto/digest"
+	"github.com/orbs-network/orbs-network-go/test"
+	"github.com/orbs-network/orbs-network-go/test/builders"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/client"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+func TestGetTransactionStatus_CallsTxPool(t *testing.T) {
+	test.WithContext(func(ctx context.Context) {
+		harness := newPublicApiHarness(ctx, 1*time.Second)
+		harness.OnGetTransactionSetPending()
+
+		harness.papi.GetTransactionStatus(&services.GetTransactionStatusInput{
+			ClientRequest: (&client.GetTransactionStatusRequestBuilder{}).Build(),
+		})
+
+		ok, err := harness.txpMock.Verify()
+		require.True(t, ok, "should have called the txp func")
+		require.NoError(t, err, "error happened when it should not")
+	})
+}
+
+func TestGetTransactionStatus_GetCommitStatusFromTxPool(t *testing.T) {
+	test.WithContext(func(ctx context.Context) {
+		harness := newPublicApiHarness(ctx, 1*time.Second)
+
+		txb := builders.Transaction().Builder()
+		txHash := digest.CalcTxHash(txb.Build().Transaction())
+
+		harness.OnGetTransactionSetCommit()
+		result, err := harness.papi.GetTransactionStatus(&services.GetTransactionStatusInput{
+			ClientRequest: (&client.GetTransactionStatusRequestBuilder{
+				Txhash: txHash,
+			}).Build(),
+		})
+
+		require.NoError(t, err, "error happened when it should not")
+		require.NotNil(t, result, "get transaction status returned nil instead of object")
+		require.Equal(t, protocol.TRANSACTION_STATUS_COMMITTED, result.ClientResponse.TransactionStatus(), "got wrong status")
+		require.NotNil(t, result.ClientResponse.TransactionReceipt(), "got empty receipt")
+	})
+}
+
+func TestGetTransactionStatus_GetPendingStatusFromTxPool(t *testing.T) {
+	test.WithContext(func(ctx context.Context) {
+		harness := newPublicApiHarness(ctx, 1*time.Second)
+
+		txb := builders.Transaction().Builder()
+		txHash := digest.CalcTxHash(txb.Build().Transaction())
+
+		harness.OnGetTransactionSetPending()
```

Comment:
these methods don't need to be exported
also, maybe name the setup method according to the scenarios they describe:
```
harness.transactionIsInPool(txHash)
harness.transactionNotInPool(txHash)
```